src/AddressSpace.cc

```diff
@@ -1040,10 +1040,16 @@ void AddressSpace::fix_stack_segment_start(const Mapping& mapping,
 }
 
 Mapping AddressSpace::vdso() const {
+  return vdso_pair().first;
+}
+
+MappingResourcePair AddressSpace::vdso_pair() const{
   assert(!vdso_start_addr.is_null());
-  return mapping_of(vdso_start_addr).first;
+  auto mapping = mapping_of(vdso_start_addr);
+  return mapping;
 }
```

Comment:
Actually it's probably a good idea to only do that if some magic environment variable is set, e.g. RR_UNMAP_VDSO_PAGE, by checking for it using `getenv()`. And then set that environment variable in a test script, e.g. copy the `hardlink_mmapped_files.run` for a new test `unmap_vdso.run`.